programs/test/dlopen.c

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Test dynamic loading of libtfpsacrypto
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "psa/build_info.h"
+#include "psa/crypto.h"
+#include "tf_psa_crypto/platform.h"
+
+#if defined(__APPLE__)
+#define SO_SUFFIX ".dylib"
+#else
+#define SO_SUFFIX ".so"
+#endif
+
+#define CRYPTO_SO_FILENAME "libtfpsacrypto" SO_SUFFIX
+#define CRYPTO_SO_PATH "core/" CRYPTO_SO_FILENAME
```

Comment:
I do not see an equivalent of that in the Mbed TLS dlopen.c (no library directory involved). Do we really need CRYPTO_SO_PATH? Could CRYPTO_SO_FILENAME be enough?

Reply:
I have found in local testing that this was required. I guess it is to do with in-tree vs. out-of-tree builds?

Alternatively I could do it like `#define CRYPTO_SO_FILENAME "core/libtfpsacrypto" SO_SUFFIX` is that is more pleasing? But as far as I have found the `core` is needed in order to run.

Reply:
Thanks for the info. That's fine, let's let it the way it is.